lib/cmd.js

```diff
@@ -28,19 +31,39 @@ Cmd.prototype.process = function(args) {
 
 Cmd.prototype.newApp = function() {
   var self = this;
-  program
-  .command('new [name]')
-  .description('new application')
-  .action(function(name, options) {
-    if (name === undefined) {
-
-      console.log("please specify your app Name".red);
-      console.log("e.g embark new MyApp".green);
-      console.log("e.g embark new --help for more information".green);
-      process.exit(9);
+
+  var validateName = function (value) {
+    try  {
+      if(value.match(/^[a-zA-Z\s\-]+$/)) return value;
+    } catch (e) {
+      throw new Error('Name must be only letters, spaces, or dashes');
     }
-    self.Embark.generateTemplate('boilerplate', './', name);
-  });
+  };
+
+  program
+    .command('new [name]')
+    .description('new application')
+    .action(function (name, options) {
+      var parentDirectory = path.dirname(__dirname).split("/").pop();
```

Comment:
I thought about that too. With events we can react and really separate this bit out.